python/across_api/base/api.py

```diff
@@ -19,3 +23,167 @@
+            title="End",
+        ),
+    ] = None,
+) -> dict:
+    """
+    Helper function to convert begin and end to datetime objects.
+    """
+    if begin is None or end is None:
+        return {"begin": None, "end": None}
+    return {"begin": Time(begin), "end": Time(end)}
+
+
+OptionalDateRangeDep = Annotated[dict, Depends(optional_daterange)]
+
+
+# Depends functions for FastAPI calls.
+async def optional_length(
+    length: Annotated[
+        Optional[float],
+        Query(
+            description="Length of time (days).",
+            title="Length",
+        ),
+    ] = None,
+) -> Optional[u.Quantity]:
+    """
+    Helper function to convert begin and end to datetime objects.
+    """
+    if length is None:
+        return None
+    return length * u.day
+
+
+OptionalLengthDep = Annotated[dict, Depends(optional_length)]
+
+
+async def optional_limit(
+    limit: Annotated[
+        Optional[int],
+        Query(
+            ge=0,
+            title="Limit",
+            description="Maximum number of results to return.",
+        ),
+    ] = None,
+) -> Optional[int]:
+    return limit
+
+
+LimitDep = Annotated[Optional[int], Depends(optional_limit)]
+
+
+async def error_radius(
+    error_radius: Annotated[
+        Optional[float],
+        Query(
+            ge=0,
+            title="Error Radius",
+            description="Error radius in degrees.",
+        ),
+    ] = None,
+) -> Optional[float]:
+    if error_radius is None:
+        return None
+    return error_radius * u.deg
+
+
+ErrorRadiusDep = Annotated[float, Depends(error_radius)]
+
+
+async def exposure(
+    exposure: Annotated[
+        float,
+        Query(
+            ge=0,
+            title="Exposure",
+            description="Exposure time in seconds.",
+        ),
+    ] = 200,
+) -> u.Quantity:
+    return exposure * u.s
+
+
+ExposureDep = Annotated[float, Depends(exposure)]
+
+
+async def offset(
+    offset: Annotated[
+        float,
+        Query(
+            ge=-200,
+            le=200,
+            title="Offset",
+            description="Offset start of dump window from T0 by this amount (seconds).",
+        ),
+    ] = -50,
+) -> u.Quantity:
+    return offset * u.s
+
+
+OffsetDep = Annotated[float, Depends(offset)]
+
+
+async def optional_ra_dec(
+    ra: Annotated[
+        Optional[float],
+        Query(
+            ge=0,
+            lt=360,
+            title="RA (J2000)",
+            description="Right Ascenscion in J2000 coordinates and units of decimal degrees.",
+        ),
+    ] = None,
+    dec: Annotated[
+        Optional[float],
+        Query(
+            ge=-90,
+            le=90,
+            title="Dec (J2000)",
+            description="Declination in J2000 coordinates in units of decimal degrees.",
+        ),
+    ] = None,
+) -> Optional[dict]:
+    if ra is None or dec is None:
+        return {"ra": None, "dec": None}
+    return {"ra": ra * u.deg, "dec": dec * u.deg}
```

Comment:
Would be more useful to return an `astropy.coordinates.SkyCoord` instance.

Reply:
This is going to lead to a bunch of unnecessary refactoring of the code, so let's address it in a later PR.

Reply:
See #2006.